hw/top_earlgrey/ip/ast/doc/_index.md

```diff
@@ -0,0 +1,1006 @@
+To activate the ADC, the corresponding [<u>comportable
+module</u>](https://docs.google.com/document/d/1_GwCccYp0zFI8OT5so4832OMEXkC-I306tSsM3i92nA/edit?usp=sharing)
+must first select the channel to sample, and then activate the ADC
+through \`adc\_pd\_i\`. Once activated, the ADC will produce an output
+synchronous to the adc controller.
+
+## ADC Usage Flow
+
+1.  > Activate the ADC by negating ‘adc\_pd\_i’
+
+2.  > Wait 30 uS for the ADC to wake up.
+
+3.  > Select an analog channel to measure by setting the corresponding
+    > bit in ‘adc\_chnsel\_i’ bus. This triggers a measurement.
+
+4.  > Wait unitl ‘adc\_d\_val’ is set and read the result via
+    > ‘adc\_d\_o’
+
+5.  > Clear ‘adc\_chnsel\_i’ bus to 0. Note that adc\_chnsel must be
+    > cleared to 0 before a new channel is selected.
+
+6.  > Repeat steps 3-5 if more channels or more measurements are
+    > required
+
+7.  > Deactivate the ADC by setting ‘adc\_pd’ to save power.
+
+![](./media/image3.png)
+
+[<u>https://wavedrom.com/editor.html</u>](https://urldefense.proofpoint.com/v2/url?u=https-3A__wavedrom.com_editor.html&d=DwMFaQ&c=ue8mO8zgC4VZ4q_aNVKt8G9MC01UFDmisvMR1k-EoDM&r=ORLxyPY9MsmpJYCWs7a9VBneW9h5vR8MnW3aYvrOooI&m=ts63KFxLdp_aXcU3h1dbe_sIHcNCW8am1RhZxv1PNbA&s=MzZVoXpb6wDp_3iGVjovwJ1b-NW3uLacU2X-WhLfAZ4&e=)
+
+{signal: \[ {node: '.a..b........', phase:0.2},
+
+{name: 'adc\_pd\_i' , wave: '10|..|.....|....|..1'},
+
+{name: 'clk\_ast\_adc\_i', wave: 'p.|..|.....|....|...'},
+
+{name: 'adc\_chnsel\_i' , wave: '0.|.3|..04.|....|0..'},
+
+{name: 'adc\_d\_val\_o' , wave: '0.|..|.1.0.|.1..|.0.'},
+
+{name: 'adc\_d\_o' , wave: 'x.|..|.3.x.|.4..|.x.', data: \['ch0', 'ch1',
+'ch1'\]},
+
+\],edge: \[ 'a\<-\>b wakeup time', \] }
+
+# Random Number Generator
+
+AST contains a random number generator that outputs random number
+bitstreams whenever it is enabled. After enabled by the [<u>comportable
+controller</u>](https://github.com/lowRISC/opentitan/blob/master/hw/ip/entropy_src/doc/_index.md)
+through ‘rng\_en\_i’, the AST begins generating multiple independent
+four random bit streams when \`rng\_ok\_o\` is asserted.
+
+The comportable controller then samples the rng bitstream at a
+configurable rate (random data is produced at \~50KHz rate). It is
+important to note that rng\_ok is NOT used to indicate when to sample,
+but simply that the rng is in a valid operational state. It is up to the
+comportable controller to sample at the correct rate to ensure a
+particular random bit is not consumed more than once. This rate will be
+known in advance and configured by software. For more information on how
+the OpenTitan entropy system operates, please see the most recent.
+
+# Entropy Consumption
+
+AST consumes entropy for defensive purposes. However, AST does not
+consume its raw entropy directly. Instead, AST receives entropy from the
+[<u>CSRNG</u>](https://github.com/lowRISC/opentitan/tree/master/hw/ip/csrng).

```

Comment:
hmmm.. we need to update this link as well later.

Reply:
OK